src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -697,58 +696,84 @@ public static long ReadLastEtag(Transaction tx)
         }
 
         public IEnumerable<Document> GetDocumentsStartingWith(DocumentsOperationContext context, string idPrefix, string matches, string exclude, string startAfterId,
-            long start, long take, string collection)
+            long start, long take, string collection, DocumentFields fields = DocumentFields.All)
         {
-            foreach (var doc in GetDocumentsStartingWith(context, idPrefix, matches, exclude, startAfterId, start, take))
+            var isAllDocs = collection == Constants.Documents.Collections.AllDocumentsCollection;
+            var requestedDataField = fields.HasFlag(DocumentFields.Data);
+            if (isAllDocs == false && requestedDataField == false)
```

Comment:
why do we need this? to read the metadat?

Reply:
Yes